.github/actions/setup/action.yml

```diff
@@ -0,0 +1,48 @@
+name: Setup
+description: Common setup steps used by our workflows
+runs:
+  using: composite
+  steps:
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: 8.6.5
+
+    - name: Setup node 16.x
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: .nvmrc
+        registry-url: https://registry.npmjs.org
+        cache: pnpm
+
+    - name: Get foundry version
+      id: foundry-version
+      run: echo ::set-output name=version::$(cat .foundryrc)
```

Comment:
@maurelian this the .foundryrc is making the devops-y part of my brain very happy this is great hermetic building here

Reply:
🫡